File: week3/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) {
            return false;
        }
        // Sort both strings as character arrays
        char[] sArr = s.toCharArray();
        Arrays.sort(sArr);
        char[] tArr = t.toCharArray();
        Arrays.sort(tArr);
        // Check whether sorted arrays are equal
        return String.valueOf(sArr).equals(String.valueOf(tArr));
    }
}

File: summer2021/unit1/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sd = self.makeDict(s)
        td = self.makeDict(t)
        return sd == td
    
    def makeDict(self, w: str) -> dict:
        result = {}
        for i in w:
            if i in result:
                result[i] += 1
            else:
                result[i] = 1
        return result